src/pages/ProfilesExplorerView/components/SceneLabelValuesHistogram.tsx

```diff
@@ -0,0 +1,146 @@
+import {
+  PanelBuilders,
+  SceneComponentProps,
+  SceneDataTransformer,
+  SceneObjectBase,
+  SceneObjectState,
+  VizPanel,
+  VizPanelState,
+} from '@grafana/scenes';
+import { SortOrder } from '@grafana/schema';
+import { LegendDisplayMode, TooltipDisplayMode, VizLegendOptions } from '@grafana/ui';
+import React from 'react';
+
+import { EventTimeseriesDataReceived } from '../domain/events/EventTimeseriesDataReceived';
+import { getColorByIndex } from '../helpers/getColorByIndex';
+import { getSeriesLabelFieldName } from '../infrastructure/helpers/getSeriesLabelFieldName';
+import { getSeriesStatsValue } from '../infrastructure/helpers/getSeriesStatsValue';
+import { buildTimeSeriesQueryRunner } from '../infrastructure/timeseries/buildTimeSeriesQueryRunner';
+import { addRefId, addStats, sortSeries } from './SceneByVariableRepeaterGrid/infrastructure/data-transformations';
+import { GridItemData } from './SceneByVariableRepeaterGrid/types/GridItemData';
+
+interface SceneLabelValuesHistogramState extends SceneObjectState {
+  body: VizPanel;
+  legendPlacement: VizLegendOptions['placement'];
+}
+
+export class SceneLabelValuesHistogram extends SceneObjectBase<SceneLabelValuesHistogramState> {
+  constructor({
+    item,
+    headerActions,
+    legendPlacement,
+  }: {
+    item: GridItemData;
+    headerActions: (item: GridItemData) => VizPanelState['headerActions'];
+    legendPlacement?: SceneLabelValuesHistogramState['legendPlacement'];
+  }) {
+    super({
+      key: 'histogram-label-values',
+      legendPlacement: legendPlacement || 'bottom',
+      body: PanelBuilders.histogram()
+        .setTitle(item.label)
+        .setData(
+          new SceneDataTransformer({
+            $data: buildTimeSeriesQueryRunner(item.queryRunnerParams),
+            transformations: [addRefId, addStats, sortSeries],
+          })
+        )
+        .setHeaderActions(headerActions(item))
+        .build(),
+    });
+
+    this.addActivationHandler(this.onActivate.bind(this, item));
+  }
+
+  onActivate(item: GridItemData) {
+    const { body } = this.state;
+
+    const sub = (body.state.$data as SceneDataTransformer)!.subscribeToState((newState) => {
+      if (newState.data?.state !== LoadingState.Done) {
+        return;
+      }
+
+      const { series } = newState.data;
+
+      if (series?.length) {
+        body.setState(this.getConfig(item, series));
+      }
+
+      // we publish the event only after setting the new config so that the subscribers can modify it
+      this.publishEvent(new EventTimeseriesDataReceived({ series }), true);
+    });
+
+    return () => {
+      sub.unsubscribe();
+    };
+  }
+
+  getConfig(item: GridItemData, series: DataFrame[]) {
+    const { legendPlacement } = this.state;
+    const groupByLabel = item.queryRunnerParams.groupBy?.label;
+
+    return {
+      title: series.length > 1 ? `${item.label} (${series.length})` : item.label,
+      options: {
```

Comment:
We might have to offer a way to tweak the bucket options (size) here because the histogram can appear empty.